CIScripts/Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -18,26 +32,36 @@ class LogSource {
 class FileLogSource : LogSource {
     [String] $Path
 
-    [Hashtable] GetContent() {
+    [CollectedLog[]] GetContent() {
         $ContentGetterBody = {
             Param([Parameter(Mandatory = $true)] [string] $From)
             $Files = Get-ChildItem -Path $From -ErrorAction SilentlyContinue
-            $Logs = @{}
+            $Logs = @()
             if (-not $Files) {
-                $Logs[$From] = "<FILE NOT FOUND>"
+                $Logs += @{
+                    Name = $From
+                    Err = "<FILE NOT FOUND>"
+                }
             } else {
                 foreach ($File in $Files) {
                     $Content = Get-Content -Raw $File
-                    $Logs[$File.FullName] = if ($Content) {
-                        $Content
-                    } else {
-                        "<FILE WAS EMPTY>"
+                    $Logs += @{
```

Comment:
Why specific type is not available here? Why do we need casting from lines 57-64? Maybe a comment in the code might be beneficial for future users.

Reply:
This scriptblock is sometimes invoked on remote machine. You're right, I'll add a comment.